### PR TITLE
Add r-rpostgres

### PR DESCRIPTION
**Pull request by halldc**
Adding a recipe for the [RPostgres](https://github.com/r-dbi/RPostgres) R package.

This was created using the [conda_r_skeleton_helper](https://github.com/bgruening/conda_r_skeleton_helper), and then `postgresql` was added to the `host` requirements and `libpq` was added to the `run` requirements.

### CODE REVIEW

**Comment by halldc** [comment]
The Linux build on CircleCI failed during testing, with the following error message:
```
> library('RPostgres')
Error: package or namespace load failed for ‘RPostgres’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/home/conda/feedstock_root/build_artifacts/r-rpostgres_1533322694527/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/R/library/RPostgres/libs/RPostgres.so':
  libgfortran.so.4: cannot open shared object file: No such file or directory
```

Following [this advice](https://github.com/conda-forge/staged-recipes/pull/3302#issuecomment-315650455), the command `conda inspect linkages r-rpostgres` returned:
```
r-rpostgres
-----------

gettext-0.19.8.1-0:
    libintl.8.dylib (lib/libintl.8.dylib)

libcxx-6.0.0-0:
    libc++.1.dylib (lib/libc++.1.dylib)

libgfortran-3.0.0-0:
    libgfortran.3.dylib (lib/libgfortran.3.dylib)

libpq-9.6.3-0:
    libpq.5.dylib (lib/libpq.5.dylib)

r-base-3.4.1-h4fe35fd_8:
    libR.dylib (lib/R/lib/libR.dylib)

r-rpostgres-1.1.1-r341h466af19_0:
    RPostgres.so (lib/R/library/RPostgres/libs/RPostgres.so)

system:
    CoreFoundation (/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation)
    libSystem.B.dylib (/usr/lib/libSystem.B.dylib)
```
and so I added `libgfortran # [not win]` to the runtime requirements.

**Comment by halldc** [comment]
I'm still seeing the same error on Linux. @bgruening do you know why this might happen?

**Comment by halldc** [comment]
@sodre Do you mean opening an issue with the [`postgresql`](https://github.com/conda-forge/postgresql-feedstock) or [`libpq`](https://github.com/conda-forge/libpq-feedstock) feedstock? I require the former at build-time, but the latter at run-time.

Why would adding `libgfortran` as a run-time dependency of `postgresql` fix this problem? I tried adding this as a dependency of `r-rpostgres` – wouldn't this be expected to fix the Linux error message?

**Comment by halldc** [comment]
I noticed this warning in the Circle-CI build log:

`WARNING (r-rpostgres,lib/R/library/RPostgres/libs/RPostgres.so): Needed DSO lib/libgfortran.so.4 found in ['libgfortran-ng']`

so I replaced the `libgfortran` dependency with a `libgfortran-ng` dependency.

Now all tests are successful. Ping @bgruening

File: recipes/r-rpostgres/meta.yaml [hunk 1]
```diff
@@ -52,7 +52,7 @@ requirements:
   run:
     - r-base
     - {{native}}gcc-libs         # [win]
-    - libgfortran                # [not win]
+    - libgfortran-ng                # [not win]
```
Comment (bgruening) [inline recipes/r-rpostgres/meta.yaml]: please use this instead if fortran is really needed `{{ compiler('fortran') }}  # [not win]` in your build section.

Comment (halldc) [reply recipes/r-rpostgres/meta.yaml]: @bgruening RPostgres doesn't need a fortran compiler itself, but PostgreSQL does. So I've removed the `libgfortran` dependency and added `postgresql` to the list of run-time dependencies. Now the builds are successful without an explicit `libgfortran` dependency.

**Comment by halldc** [comment]
@conda-forge/staged-recipes 

I believe this is now ready to merge – all builds are passing.

I spent a long time understanding how the `libgfortran` and `libfortran-ng` packages were conflicting (by having both available at build-time and only one available at run-time). But in the end, it seems that the `postgresql` package has recently been rebuilt and this was no longer an issue.

**Comment by bgruening** [comment]
Thanks @halldc for investigating!

**Pull request merged**